src/ansys/aedt/core/modeler/circuits/primitives_maxwell_circuit.py

```diff
@@ -284,3 +284,87 @@ def create_winding(self, name=None, location=None, angle=0, use_instance_id_netl
         )
         id.set_property("Name", name)
         return id
+
+    @pyaedt_function_handler(compname="name")
+    def create_ISin(self, name=None, value=1, location=None, angle=0, use_instance_id_netlist=False):
+        """Create a sinusoidal current source.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the current source. The default is ``None``.
+        value : float, optional
+            Value for the amplitude of current. The default is ``1``.
+        location : list of float, optional
+            Position on the X axis and Y axis.
+        angle : float, optional
+            Angle of rotation in degrees. The default is ``0``.
+        use_instance_id_netlist : bool, optional
+            Whether to use the instance ID in the net list or not. The default is ``False``.
+
+        Returns
+        -------
+        :class:`ansys.aedt.core.modeler.cad.object_3dcircuit.CircuitComponent`
+            Circuit Component Object.
+
+        References
+        ----------
+
+        >>> oEditor.CreateComponent
+        """
+        if location is None:
+            location = []
+        id = self.create_component(
```

Comment:
Please avoid using `id` as it is a built-in function.
![image](https://github.com/user-attachments/assets/50f5bda5-18bf-484f-84fa-2ef4e7891f18)

Reply:
OK, I'll adjust it to 'compid', I'll also change the other five existing uses in this file as well whilst I'm there

Reply:
@DaveTwyman please also remember to adhere to python case convention and use underscores to separate words.

Reply:
@gmalinve , Ok will do, just to check this is referring to the 'compid' suggestion, if so I'll use 'comp_id' instead

Reply:
@SMoraisAnsys ,@gmalinve
Yikes, looking at this a bit deeper, 'id' is also the returned value from many of the existing circuit components in both 'primitives_circuit' and 'primitives_maxwell_circuit'. Probably best if I stick to using the existing convention for now as it works. We could raise another pull request to make all these changes another time?

Reply:
@DaveTwyman we can think about this later :) first you need to fix your methods and add unit tests

Reply:
Yes, I agree